evm_arithmetization/src/cpu/control_flow.rs

```diff
@@ -78,6 +78,16 @@ pub(crate) fn eval_packed_generic<P: PackedField>(
     );
     yield_constr.constraint_transition(is_prover_input * (lv.is_kernel_mode - nv.is_kernel_mode));
 
+    // Check the helper value in the general columns.
+    yield_constr.constraint(
+        lv.op.push_prover_input
+            * ((lv.is_kernel_mode + lv.general.push().push_prover_input_not_kernel) - P::ONES),
+    );
+
+    // The general column should be not be set when the operation is a
+    // `PROVER_INPUT`.
+    yield_constr.constraint(is_prover_input * lv.general.push().push_prover_input_not_kernel);
```

Comment:
I think it's luckily already checked by the previous constraint since `PROVER_INPUT` is a kernel-only instruction.

Reply:
The previous constraint is just ensuring that we don't change mode between current and next row. Did you mean this is already checked by the `decode` module instead? (in which case I agree, it is redundant, I'll remove it but add a comment to refer to the constraint there).

Reply:
Ah, I had misread your comment. We're aligned then